lib/pages/classes/service/class_provider.dart

```diff
@@ -287,4 +290,22 @@ class ClassProvider with ChangeNotifier {
       return false;
     }
   }
+
+  Future<List<ClassHeader>> search(String query) async {
+    final List<ClassHeader> classes = await fetchClassHeaders();
+    final List<String> searchedWords = query
+        .toLowerCase()
+        .split(' ')
+        .where((element) => element != '')
+        .toList();
+    return classes.where(
```

Comment:
Nice use of functional structures. Though the formatting is really weird here and in the rest of the file, please fix.